### ISSUE TITLE
FindAll In Range

### ISSUE THREAD

**Issue by ArunPatal** [ajaxorg/ace#4624]
``Hello,

I was trying to get some data within range but find all somehow ignores column.
I hope i am using findAll right...

This is findAll function...

```
var FindAll = function () {
	var editor = ace.edit("Demo-Editor");

	var options = {
		range: {start: {row: 0, column: 0}, end: {row: 1, column: 22}}
	};

	editor.$search.setOptions({
		needle: "var",
		backwards: false, // Whether to search backwards from where cursor currently is. Defaults to `false`
		regExp: false, // Whether the search is a regular expression or not. Defaults to `false`
		range: options.range, // The [[Range]] to search within. Set this to `null` for the whole document
		wrap: false, // Whether to wrap the search back to the beginning when it hits the end. Defaults to `false`
		caseSensitive: true, // Whether the search ought to be case-sensitive. Defaults to `false`
		skipCurrent: false, // Whether or not to include the current line in the search. Default to `false`
		wholeWord: true, // Whether the search matches only on whole words. Defaults to `false`
		start: options.range.start // From where search show start
	});
 
	var searchData = editor.$search.findAll(editor.session);
	console.log(searchData.length, JSON.stringify(searchData));

};
```

And this is html

```
<div id="Demo-Editor">
	var myVar1 = 1; var myVar2 = 2; var myVar3 = 3;
	var myVar4 = 4; var myVar5 = 5; var myVar6 = 6;
</div>
```

According to range i should also get var before myVar2. But i am getting only one object in searchData array.

**Comment by ArunPatal** [comment]
I also have created a demo..
[FindAll Test](https://codepen.io/arunpatal/pen/NWjOowJ?editors=1111)

**Comment by andrewnester** [comment]
Yes, indeed at the moment we only respect `row` field, see 
https://github.com/ajaxorg/ace/blob/4257621787b4253d6d493611f4ec5a37829da323/lib/ace/search.js#L141

We either need to fix this or be more explicit about it in our documentation

**Comment by ArunPatal** [comment]
I request to fix this issue as it can be very useful for some cases... Thanks

**Comment by andrewnester** [comment]
Fixed and released in 1.12.2
